sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -1607,6 +1796,10 @@ private synchronized void flushBatch()
         CommitRequest.Builder commitRequest = CommitRequest.newBuilder();
         commitRequest.addAllMutations(mutations);
         commitRequest.setMode(CommitRequest.Mode.NON_TRANSACTIONAL);
+        commitRequest.setProjectId(projectId.get());
+        commitRequest.setDatabaseId(databaseId);
+        commitRequest.setProjectId(projectId.get());
```

Comment:
does this need to happen twice?

Reply:
It was caused by a bad rebase. Good catch.